packages/editor/src/store/actions.js

```diff
@@ -102,43 +195,276 @@ export function editPost( edits ) {
+				isRevision: newPost.id !== post.id,
+			}
+		);
+
+		const notifySuccessArgs = getNotificationArgumentsForSaveSuccess( {
+			previousPost: post,
+			post: newPost,
+			postType,
+			options,
+		} );
+		if ( notifySuccessArgs.length > 0 ) {
+			yield dispatch(
+				'core/notices',
+				'createSuccessNotice',
+				...notifySuccessArgs
+			);
+		}
+	} catch ( error ) {
+		yield dispatch(
+			MODULE_KEY,
+			'__experimentalRequestPostUpdateFailure',
+			{ post, edits, error, options }
+		);
+		const notifyFailArgs = getNotificationArgumentsForSaveFail( {
+			post,
+			edits,
+			error,
+		} );
+		if ( notifyFailArgs.length > 0 ) {
+			yield dispatch(
+				'core/notices',
+				'createErrorNotice',
+				...notifyFailArgs
+			);
+		}
+	}
 }
 
-export function trashPost( postId, postType ) {
-	return {
-		type: 'TRASH_POST',
-		postId,
-		postType,
-	};
+/**
+ * Action generator for handling refreshing the current post.
+ */
+export function* refreshPost() {
+	const post = yield select(
+		MODULE_KEY,
+		'getCurrentPost'
+	);
+	const postTypeSlug = yield select(
+		MODULE_KEY,
+		'getCurrentPostType'
+	);
+	const postType = yield resolveSelect(
+		'core',
+		'getPostType',
+		postTypeSlug
+	);
+	const newPost = yield apiFetch(
+		{
+			// Timestamp arg allows caller to bypass browser caching, which is
+			// expected for this specific function.
+			path: `/wp/v2/${ postType.rest_base }/${ post.id }?context=edit&_timestamp=${ Date.now() }`,
+		}
+	);
+	yield dispatch(
+		MODULE_KEY,
+		'resetPost',
+		newPost
+	);
+}
+
+/**
+ * Action generator for trashing the current post in the editor.
+ */
+export function* trashPost() {
+	const postTypeSlug = yield select(
+		MODULE_KEY,
+		'getCurrentPostType'
+	);
+	const postType = yield resolveSelect(
+		'core',
+		'getPostType',
+		postTypeSlug
+	);
+	yield dispatch(
+		'core/notices',
+		'removeNotice',
+		TRASH_POST_NOTICE_ID
+	);
+	try {
+		const post = yield select(
+			MODULE_KEY,
+			'getCurrentPost'
+		);
+		yield apiFetch(
+			{
+				path: `/wp/v2/${ postType.rest_base }/${ post.id }`,
+				method: 'DELETE',
+			}
+		);
+
+		// TODO: This should be an updatePost action (updating subsets of post
+		// properties), but right now editPost is tied with change detection.
+		yield dispatch(
+			MODULE_KEY,
+			'resetPost',
+			{ ...post, status: 'trash' }
+		);
+	} catch ( error ) {
+		yield dispatch(
+			'core/notices',
+			'createErrorNotice',
+			...getNotificationArgumentsForTrashFail( { error } ),
+		);
+	}
 }
 
 /**
- * Returns an action object used in signalling that the post should autosave.
+ * Action generator used in signalling that the post should autosave.
  *
  * @param {Object?} options Extra flags to identify the autosave.
- *
- * @return {Object} Action object.
  */
-export function autosave( options ) {
-	return savePost( { isAutosave: true, ...options } );
+export function* autosave( options ) {
+	yield* generatorActions.savePost( { isAutosave: true, ...options } );
```

Comment:
Could this be simplified to just `yield* savePost( { /* ... */ } )j` ?

Reply:
I used `generatorActions` here so I could mock the `savePost` action more easily.  It's possible I'm overlooking a way to more easily mock this in the tests or the mocking is unnecessary (we could just check that it yields a generator instead).

Reply:
saw the subsequent comment about `generatorActions` and realize that ya I can just yield savePost directly I think here.

Reply:
Won't work, need to keep this as is so it can be tested (see comment in other thread).